### PR TITLE
feat(core): switch database to Redb

### LINKED ISSUE #738: Replace Libmdbx with an alternative

### PR DESCRIPTION
**Pull request by jrchatruc**
**Motivation**

**Description**

Closes #738

### CODE REVIEW

File: crates/storage/store/engines/libmdbx.rs [hunk 1]
```diff
@@ -580,6 +609,45 @@ impl Encodable for ChainDataIndex {
     }
 }
 
+impl redb::Value for ChainDataIndex {
```
Comment (fmoletta) [inline crates/storage/store/engines/libmdbx.rs:612]: We shouldn't have code necessary for the redb impl in a module gated behind the `libmdbx` feature, maybe we could move `ChainDataIndex` outside this module ( to engine or engine/utils) and only keep implementation-specific code here

**Review by fmoletta — COMMENTED** [review]
In the storage module we have tests that run the full test suite against the different engine implementations, we should add one for redb too. We just need to add this test:
```
#[test]
    fn test_redb_store() {
        test_store_suite(EngineType::Redb);
    }
```

File: crates/storage/store/engines/libmdbx.rs (hunk 1 above)
Comment (jrchatruc) [reply crates/storage/store/engines/libmdbx.rs]: Agreed, I moved the `ChainDataIndex` enum to its own `utils` module under `engines` here [dbf7bdd](https://github.com/lambdaclass/ethrex/pull/1351/commits/dbf7bdd9375f1f2f2b8284f859eddb55a2ed6c74)

**Comment by jrchatruc** [comment]
> In the storage module we have tests that run the full test suite against the different engine implementations, we should add one for redb too. We just need to add this test:
> 
> ```
> #[test]
>     fn test_redb_store() {
>         test_store_suite(EngineType::Redb);
>     }
> ```

Nice!, added the test here [404f388](https://github.com/lambdaclass/ethrex/pull/1351/commits/404f388536d54ae086c900833d6993f8ebb74e9b)

**Review by ilitteri — APPROVED** [review]
LGTM

**Pull request merged**